### DISCUSSION TITLE
Getter for Scenario to set default Scenarios

### DISCUSSION THREAD

**Discussion by dschibster** [jongpie/NebulaLogger#278] [Ideas]
Hi there @jongpie!

I am in the middle of cleanup project, and want to get to use Nebula Logger to finally get some transparency about when automation go awry. One thing I want to use extensively is the scenario feature. However, I think that when you enter Apex from a Component, it's pretty easy to set one, but not so much when you run a basic update from UI. The thing is, I was thinking if setting a Default Scenario in these cases would be wise, which I wouldn't be able to do right now because I wouldn't be able to inspect the Scenario at runtime without fiddling in the code.

What do you think about the option of adding a Default Scenario, and what would you say about adding a getter to check if we need to set one in our Triggers?

**Comment by jongpie** [comment]
Hi @dschibster - that seems reasonable to me! When I originally added the method `Logger.setScenario(String)`, I had thought about also adding a method `Logger.getScenario()`, but held off since it wasn't part of the original request - I'm happy to go ahead and add that method as well.

As far as a Default Scenario field, I think it would make sense for that to be a new field on `LoggerSettings__c` so you can set the default at the org, profile or user-level.

I've created a new issue, #279, to track this enhancement. I think I can included it in the next release, `v4.7.0`, that I'm hoping to have released sometime next week (see PR #274). I'll keep you updated on the status!

**Reply by dschibster** [reply]
That would be lovely! Thanks for the quick response.
